expbackoff/backoff.go

```diff
@@ -0,0 +1,100 @@
+package expbackoff
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+// BackOff is wrapper that provides retry mechanism.
+type BackOff struct {
+	counter prom.Counter
+	backoff backoff.BackOff
+	timeout time.Duration
+}
+
+type RetriesCfg struct {
+	Limited bool
+	Limit   uint64
+}
+
+// New instance of BackoffTimeout.
+func New(
+	ctx context.Context,
+	counter prom.Counter,
+	timeout time.Duration,
+	retriesCfg RetriesCfg,
+	opts ...Option,
+) *BackOff {
+	var expBackoff backoff.BackOff
+	expBackoff = backoff.WithContext(new(opts...), ctx)
```

Comment:
expBackoff :=

Reply:
`var expBackoff backoff.BackOff` required here: `WithContext` and `WithMaxRetires` returns ptrs to different structs that implements same `Backoff` interface.

Otherwise it won't complile:
```
cannot use backoff.WithMaxRetries(expBackoff, retriesCfg.Limit) (value of type "github.com/cenkalti/backoff/v4".BackOff) as "github.com/cenkalti/backoff/v4".BackOffContext value in assignment: missing method Contextcompiler[InvalidIfaceAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#InvalidIfaceAssign)
```